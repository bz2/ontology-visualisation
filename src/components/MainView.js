import { connect } from 'redux-zero/react'
import PropTypes from 'prop-types'
import Sidebar from './Sidebar'
import GraphVisualisation from './GraphVisualisation'
import ElementInfo from './ElementInfo'
import SearchBox from './SearchBox'
import SettingsBox from './SettingsBox'
import EdgeFilter from './EdgeFilter'

const MainView = ({
  isSidebarOpen,
  isSearchOpen,
  isInfoOpen,
  selectedNode,
  isEdgeFilterOpen,
  isSettingsOpen
}) => (
  <main className="main-view">
    {
      isSidebarOpen && (
        <Sidebar />
      )
    }

    <div className="main-view-area">
      {
        isInfoOpen
        && selectedNode
        && (
          <ElementInfo />
        )
      }

      {
        isSearchOpen
        && (
          <SearchBox />
        )
      }

      <GraphVisualisation />

      {
        isEdgeFilterOpen
        && (
          <EdgeFilter />
        )
      }

      {
        isSettingsOpen
        && (
          <SettingsBox />
        )
      }
    </div>
  </main>
)

MainView.propTypes = {
  isSidebarOpen: PropTypes.bool.isRequired,
  isSearchOpen: PropTypes.bool.isRequired,
  isInfoOpen: PropTypes.bool.isRequired,
  selectedNode: PropTypes.string,
  isEdgeFilterOpen: PropTypes.bool.isRequired,
  isSettingsOpen: PropTypes.bool.isRequired,
}

MainView.defaultProps = {
  selectedNode: undefined,
}

const mapToProps = ({
  isSidebarOpen,
  isInfoOpen,
  isSearchOpen,
  selectedNode,
  isEdgeFilterOpen,
  isSettingsOpen
}) => ({
  isSidebarOpen,
  isInfoOpen,
  isSearchOpen,
  selectedNode,
  isEdgeFilterOpen,
  isSettingsOpen
})

export default connect(
  mapToProps
)(MainView)
