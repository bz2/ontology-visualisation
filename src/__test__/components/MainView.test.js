import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import MainView from '../../components/MainView'

const setup = () => {
  const props = {
    isSidebarOpen: true,
    isSearchOpen: true,
    isNodeSelectable: true,
    selectedNodes: ['abc'],
    selectedEdges: ['123'],
    isEdgeSelectable: true,
    isSettingsOpen: true,
    isNetworkLoading: true,
    networkLoadingProgress: 10
  }

  const component = shallow(<MainView {...props} />)

  return {
    component,
    props
  }
}

describe('MainView', () => {
  it('should match snapshot ', () => {
    const {
      component
    } = setup()

    expect(toJson(component)).toMatchSnapshot()
  })
})
