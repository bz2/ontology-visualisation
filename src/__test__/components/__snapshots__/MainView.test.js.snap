// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainView should match snapshot  1`] = `
<Connect
  actions={Object {}}
  isEdgeFilterOpen={true}
  isInfoOpen={true}
  isSearchOpen={true}
  isSettingsOpen={true}
  isSidebarOpen={true}
  mapToProps={[Function]}
  selectedNode="abc"
>
  <Component />
</Connect>
`;
